src/Components/Shared/src/WebEventData.cs

```diff
@@ -6,21 +6,42 @@
 using System;
 using System.Diagnostics.CodeAnalysis;
 using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Text.Json.Serialization.Metadata;
+using Microsoft.AspNetCore.Components;
 using Microsoft.AspNetCore.Components.RenderTree;
-using static Microsoft.AspNetCore.Internal.LinkerFlags;
+using Microsoft.AspNetCore.Components.Web;
+
+[assembly: JsonSerializable(typeof(WebEventDescriptor))]
+[assembly: JsonSerializable(typeof(EventArgs))]
+[assembly: JsonSerializable(typeof(ChangeEventArgs))]
+[assembly: JsonSerializable(typeof(ClipboardEventArgs))]
+[assembly: JsonSerializable(typeof(DragEventArgs))]
+[assembly: JsonSerializable(typeof(ErrorEventArgs))]
+[assembly: JsonSerializable(typeof(FocusEventArgs))]
+[assembly: JsonSerializable(typeof(KeyboardEventArgs))]
+[assembly: JsonSerializable(typeof(MouseEventArgs))]
+[assembly: JsonSerializable(typeof(PointerEventArgs))]
+[assembly: JsonSerializable(typeof(ProgressEventArgs))]
+[assembly: JsonSerializable(typeof(TouchEventArgs))]
+[assembly: JsonSerializable(typeof(WheelEventArgs))]
```

Comment:
I thought we now had support for putting this attribute on the type itself. Is that not the case? Or do you consider it preferable to put it on the assembly for some reason?

Reply:
Following API review on Friday (https://github.com/dotnet/runtime/issues/51945#issuecomment-841456779). We're updating the design such that the user provides a partial context class as a kick off for source generation, rather than generating one for them out of the ether. This will come in preview 5, so it might be better for this PR to wait (to get new bits and also so we can get your feedback)...

> - Instead of generating a single JsonContext type per assembly, we should let the user define the JsonContext type (controlling namespace, type name, etc.) and apply the assembly level attributes to this type and have the generator fill in the body of the body. This allows the user to have multiple different JSON options for different types in a single assembly, e.g.
Fabrikam.Telemetry.TelemetryJsonContext
Fabrikam.Shopping.ShoppingJsonContext

With this model, your context definition might look like this:

```cs
[JsonSerializable(typeof(WebEventDescriptor))]
[JsonSerializable(typeof(EventArgs))]
[JsonSerializable(typeof(ChangeEventArgs))]
...
partial class WebEventDataJsonContext: JsonSerializerContext {}
```

The `JsonSerializable` attribute is now applied to context `class`es only. If we allow it directly on a contract type, then we'd have to include generated source for the type in every context definition across all assemblies. I imagine this would be undesirable in many scenarios.

Reply:
Oh neat. I might wait for this update to land before merging this change in. 

> This will come in preview 5

Did you mean preview 6 since p5 locks down soon

Reply:
> Did you mean preview 6 since p5 locks down soon

It's easier for me to include it in the PR for https://github.com/dotnet/runtime/issues/51945 which is targeted for P5. I've notified folks on the runtime end that I might need a couple more days and go through ask-mode to backport it.